src/model/test/CoilSystemCoolingWater_GTest.cpp

```diff
@@ -26,72 +38,299 @@ TEST_F(ModelFixture, CoilSystemCoolingWater_GettersSetters) {
+  EXPECT_FALSE(coilSystemClone.inletModelObject());
+  EXPECT_FALSE(coilSystemClone.outletModelObject());
+
+  EXPECT_TRUE(coilSystemClone.coolingCoil().optionalCast<CoilSystemCoolingWaterHeatExchangerAssisted>());
+  EXPECT_NE(coilSystemHXAssisted.handle(), coilSystemClone.coolingCoil().handle());
+
+  ASSERT_TRUE(coilSystemClone.companionCoilUsedForHeatRecovery());
+  EXPECT_TRUE(coilSystemClone.companionCoilUsedForHeatRecovery()->optionalCast<CoilCoolingWater>());
+  EXPECT_NE(companionCoil.handle(), coilSystemClone.companionCoilUsedForHeatRecovery()->handle());
+
+  // Clone into another Model
+  {
+    Model m2;
+    auto coilSystemClone = coilSystem.clone(m2).cast<CoilSystemCoolingWater>();
+
+    EXPECT_EQ(1, m2.getConcreteModelObjects<CoilSystemCoolingWater>().size());
+    EXPECT_EQ(1, m2.getConcreteModelObjects<CoilSystemCoolingWaterHeatExchangerAssisted>().size());
+    EXPECT_EQ(2, m2.getConcreteModelObjects<CoilCoolingWater>().size());
+    EXPECT_EQ(0, m2.getConcreteModelObjects<ControllerWaterCoil>().size());
+
+    EXPECT_FALSE(coilSystemClone.airLoopHVAC());
+    EXPECT_FALSE(coilSystemClone.inletModelObject());
+    EXPECT_FALSE(coilSystemClone.outletModelObject());
+
+    EXPECT_TRUE(coilSystemClone.coolingCoil().optionalCast<CoilSystemCoolingWaterHeatExchangerAssisted>());
+    EXPECT_NE(coilSystemHXAssisted.handle(), coilSystemClone.coolingCoil().handle());
+
+    ASSERT_TRUE(coilSystemClone.companionCoilUsedForHeatRecovery());
+    EXPECT_TRUE(coilSystemClone.companionCoilUsedForHeatRecovery()->optionalCast<CoilCoolingWater>());
+    EXPECT_NE(companionCoil.handle(), coilSystemClone.companionCoilUsedForHeatRecovery()->handle());
+
+    // Remove
+    coilSystemClone.remove();
+    EXPECT_EQ(0, m2.getConcreteModelObjects<CoilSystemCoolingWater>().size());
+    EXPECT_EQ(0, m2.getConcreteModelObjects<CoilSystemCoolingWaterHeatExchangerAssisted>().size());
+    EXPECT_EQ(0, m2.getConcreteModelObjects<CoilCoolingWater>().size());
+  }
+
+  coilSystemClone.remove();
+  EXPECT_EQ(1, m.getConcreteModelObjects<CoilSystemCoolingWater>().size());
+  EXPECT_EQ(1, m.getConcreteModelObjects<CoilSystemCoolingWaterHeatExchangerAssisted>().size());
+  EXPECT_EQ(2, m.getConcreteModelObjects<CoilCoolingWater>().size());
+  EXPECT_EQ(0, m.getConcreteModelObjects<ControllerWaterCoil>().size());
+
+  coilSystem.remove();
+  EXPECT_EQ(0, m.getConcreteModelObjects<CoilSystemCoolingWater>().size());
+  EXPECT_EQ(0, m.getConcreteModelObjects<CoilSystemCoolingWaterHeatExchangerAssisted>().size());
+  EXPECT_EQ(0, m.getConcreteModelObjects<CoilCoolingWater>().size());
+  EXPECT_EQ(0, m.getConcreteModelObjects<ControllerWaterCoil>().size());
+
+  EXPECT_EQ(3, a.supplyComponents().size());       // o --- oa_sys --- o
+  EXPECT_EQ(1, oa_sys.oaComponents().size());      // o
+  EXPECT_EQ(1, oa_sys.reliefComponents().size());  // o
+  EXPECT_EQ(5, p.demandComponents().size());       // o --- Splitter --- o --- Mixer --- o
+}
```

Comment:
Extensive test for clone and remove. Lots were going wrong, and p.demandComponents was 0 because the connections were broken.